File: src/main/java/com/aksenova/additivesmanager/dto/ProductDto.java
package com.aksenova.additivesmanager.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.Data;

import java.time.LocalDate;
import java.util.List;

@Data
public class ProductDto {

    @NotBlank(message = "productName is required")
    private String productName;

    @NotNull(message = "productTypeId is required")
    private Integer productTypeId;

    @NotNull(message = "statusId is required")
    private Integer statusId;

    private String tnVedCode;
    private String eNumber;
    private String releaseForm;
    private List<Integer> manufacturerIds;
    private String sgrNumber;
    private LocalDate sgrRegistrationDate;

    @Positive(message = "shelfLifeMonths must be positive")
    private Integer shelfLifeMonths;

    private String storageConditions;
}

File: frontend/RecipeApp.WinForms/Models/ProductDto.cs
namespace RecipeApp.WinForms.Models;

public class ProductDto
{
    public string ProductName { get; set; } = string.Empty;
    public int ProductTypeId { get; set; }
    public int StatusId { get; set; }
    public string? TnVedCode { get; set; }
    public string? ENumber { get; set; }
    public string? ReleaseForm { get; set; }
    public List<int>? ManufacturerIds { get; set; }
    public string? SgrNumber { get; set; }
    public string? SgrRegistrationDate { get; set; }
    public int? ShelfLifeMonths { get; set; }
    public string? StorageConditions { get; set; }
}
